wsagent/che-core-api-git/src/main/java/org/eclipse/che/api/git/GitUrlUtils.java

```diff
@@ -17,16 +17,64 @@
  *
  * @author Vladyslav Zhukovskii
  * @author Kevin Pollet
+ * @author Igor Vinokur
  */
 public class GitUrlUtils {
     public static final Pattern GIT_SSH_URL_PATTERN =
             Pattern.compile("((((git|ssh)://)(([^\\\\/@:]+@)??)[^\\\\/@:]+)|([^\\\\/@:]+@[^\\\\/@:]+))(:|/)[^\\\\@:]+");
+    public static final Pattern GIT_URL_WITH_CREDENTIALS_PATTERN = Pattern.compile("https?://[^:]+:.+@.*");
 
     private GitUrlUtils() {
 
     }
 
+    /**
+     * Check that given url is ssh url.
+     *
+     * @param url
+     *         git url
+     * @return {@code true} if given url is ssh url or {@code false} if not
+     */
     public static boolean isSSH(String url) {
         return url != null && GIT_SSH_URL_PATTERN.matcher(url).matches();
     }
+
+    /**
+     * Check that given url contains credentials.
+     *
+     * @param url
+     *         git url
+     * @return {@code true} if given url contains credentials or {@code false} if not
+     */
+    public static boolean containsCredentials(String url) {
```

Comment:
Are you sure that it should be a part of API or maybe it would be better to make it part of jgit implementation?
Then it will looks more simple, just one block
```
if (url != null && GIT_URL_WITH_CREDENTIALS_PATTERN.matcher(url).matches()) {
    String username = url.substring(url.lastIndexOf(":") + 1, url.indexOf("@"));
    String password = url.substring(url.lastIndexOf(":") + 1, url.indexOf("@"));
    command.setCredentialsProvider(new UsernamePasswordCredentialsProvider(username, password));
}
```